components/collection/activity/events/Events.vue

```diff
@@ -118,28 +115,23 @@ const getEventAddresses = (event): string[] => {
 
 const eventsAddresses = computed(() => {
   const addresses = props.events.map(getEventAddresses).flat()
-
-  return [...new Set([...addresses])].map(getIdentityId)
+  return [...new Set([...addresses])]
 })
 
-const { data: identities, refetch: getIdentities } = useGraphql({
-  clientName,
-  queryName: 'identities',
-  disabled: computed(() => true),
-  variables: { where: {} },
-})
+const cacheProfilesKey = computed(
+  () => `profiles-${eventsAddresses.value.sort().join(',')}`,
+)
 
-watch(
-  eventsAddresses,
+const { data: profiles } = useAsyncData(
+  cacheProfilesKey.value,
   () => {
-    getIdentities({
-      where: {
-        id_in: eventsAddresses.value,
-        display_isNull: false,
-      },
-    })
+    if (!eventsAddresses.value.length) {
+      return Promise.resolve([])
+    }
+    const cacheData = useNuxtData(cacheProfilesKey.value).data.value
```

Comment:
didn't know this was needed.
i thought by giving `useAsyncData` a key the cached data is returned automatically


anyway, works well :)